src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/IMSBuildAPIUtility.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal interface IMSBuildAPIUtility
+    {
+        public ILogger Logger { get; }
+
+        /// <summary>
+        /// Opens an MSBuild.Evaluation.Project type from a csproj file.
+        /// </summary>
+        /// <param name="projectCSProjPath">CSProj file which needs to be evaluated</param>
+        /// <returns>MSBuild.Evaluation.Project</returns>
+        Project GetProject(string projectCSProjPath);
+
+        /// <summary>
+        /// Enumerates projects within a solution
+        /// </summary>
+        /// <param name="solutionPath">Solution file whose projects need to be enumerated</param>
+        IEnumerable<string> GetProjectsFromSolution(string solutionPath);
```

Comment:
In my opinion, returning `IEnumerable` is a code smell that harms performance. If the caller ever wants to know how many items are in the collection, they'll have to use LINQ's `Count()` method, which will enumerate the enumeration, or call `ToList()` or `ToArray()` first, both of which have multiple allocations.

It's really unfortunately that so much of NuGet.Client's code base already uses methods that return `IEnumerable`, but the only valid time to use it, in my opinion, is when the method uses `yield return`. Otherwise, return a more specific data type. I wish I could find a source, but I remember a quote along the lines of "be specific in what you return, be liberal in what you accept" (IEnumerable is fine as a method argument, assuming it's only enumerated once).